tripper/backends/ontopy.py

```diff
@@ -51,73 +51,86 @@ def __init__(
     def triples(self, triple: "Triple") -> "Generator":
         """Returns a generator over matching triples."""
 
-        def to_literal(o, d):
-            """Returns a literal from (o, d)."""
-            if isinstance(d, str) and d.startswith("@"):
-                lang, datatype = d[1:], None
+        def to_literal(object_, datum):
```

Comment:
s, p, o are standard variable names when you are dealing with triples. Writing them out just makes the code lines longer and less readable. 

The `d` stands for "datatype" not "datum". The notation for language strings are e.g. "@en" for an English language string.

Reply:
The fact the I didn't know what `d` stands for speaks to the difficulty of using such variable names.
I will not be the only one not knowing this, nor knowing what `s`, `p`, and `o` stands for.

However, what we can do (since I also don't like having to add an underscore to `object`) is to write the naming convention into the module doc-string and include it in each of the function/method doc-strings as well (just copy-pasting). And also add `s`, `p`, and `o` as accepted variable names to the pylint configuration.

I would still prefer to _not_ use `d`, but instead use `datatype`.

Reply:
To quote "Clean Code" regarding single-variable names:

> Single-letter names and numeric constants have a particular problem in that they are not
> easy to locate across a body of text.
> 
> One might easily grep for `MAX_CLASSES_PER_STUDENT`, but the number 7 could be more
> troublesome. Searches may turn up the digit as part of file names, other constant definitions,
> and in various expressions where the value is used with different intent. It is even
> worse when a constant is a long number and someone might have transposed digits,
> thereby creating a bug while simultaneously evading the programmer’s search.
> 
> Likewise, the name e is a poor choice for any variable for which a programmer might
> need to search. It is the most common letter in the English language and likely to show up
> in every passage of text in every program. In this regard, longer names trump shorter
> names, and any searchable name trumps a constant in code.
> 
> My personal preference is that single-letter names can ONLY be used as local variables
> inside short methods. **The length of a name should correspond to the size of its scope**
> 
> If a variable or constant might be seen or used in multiple places in a body of code,
> it is imperative to give it a search-friendly name. Once again compare:
> 
> ```
> for (int j=0; j<34; j++) {
> s += (t[j]*4)/5;
> }
> ```
> 
> to 
> 
> ```
> int realDaysPerIdealDay = 4;
> const int WORK_DAYS_PER_WEEK = 5;
> int sum = 0;
> for (int j=0; j < NUMBER_OF_TASKS; j++) {
> int realTaskDays = taskEstimate[j] * realDaysPerIdealDay;
> int realTaskWeeks = (realdays / WORK_DAYS_PER_WEEK);
> sum += realTaskWeeks;
> }
> ```
> 
> Note that sum, above, is not a particularly useful name but at least is searchable. The
> intentionally named code makes for a longer function, but consider how much easier it
> will be to find `WORK_DAYS_PER_WEEK `than to find all the places where 5 was used and filter
> the list down to just the instances with the intended meaning.

Reply:
"Code Complete" cites interesting emperical studies: 

> Gorla, Benander, and Benander found that the effort required to debug a program was minimized when variables had names that averaged 10 to 16 characters (1990). Programs with names averaging 8 to 20 characters were almost as easy to debug. **The guideline doesn’t mean that you should try to make all of your variable names 9 to 15 or 10 to 16 characters long. It does mean that if you look over your code and see many names that are shorter, you should check to be sure that the names are as clear as they need to be.**

> Are short variable names always bad? No, not always. When you give a variable a short
> name like i, the length itself says something about the variable—namely, that the variable is a scratch value with a limited scope of operation.
> 
> A programmer reading such a variable should be able to assume that its value isn’t
> used outside a few lines of code. When you name a variable i, you’re saying, “This variable is a run-of-the-mill loop counter or array index and doesn’t have any significance
> outside these few lines of code.”
> 
> A study by W. J. Hansen found that longer names are better for rarely used variables or
> global variables and shorter names are better for local variables or loop variables

Reply:
Indeed - but in certain domains I also agree with @jesper-friis that using "standard" conventions makes things easier. In the ontology/triple domain, `s`, `p`, and `o` are frequently used as short-names for subject, predicate, and object (the three parts of a triple).
Hence, when they come together, it should ("should" being the key word here) be clear what is meant for other coders within the same context domain. For those not in the same context domain or just learning, we can support them by writing this convention in doc-strings (for the module, a dedicated documentation page or in function/method/class doc-strings).

However, this domain convention arguments falls apart for the `d` parameter in my view, hence my comment above.

Coming back to using `s`, `p`, and `o` "when they come together", raises the issue of: How "degraded" does this togetherness have to be before we start using their full names?
I.e., for the methods that take a subject and returns the predicates and objects for said subject - is this togetherness broken? What about the methods that take two out of three and provides the last breaking the togetherness?

Reply:
This is a good/interesting point to think about. What is the **scope** of these s, p, o variables? Do they frequently show up in small functions, limited loops, short methods, etc.? If so, and especially if used consistently, and in line with 'the standards' of the field, it can make sense to give them shorter names. Ad-hock labeling data/datatype/datum as d, probably not

Reply:
> I would still prefer to _not_ use `d`, but instead use `datatype`.

I bye that one, I has nothing against replacing `d` with `datatype` as long as we keep s, p, o.

Reply:
> This is a good/interesting point to think about. What is the **scope** of these s, p, o variables? Do they frequently show up in small functions, limited loops, short methods, etc.? If so, and especially if used consistently, and in line with 'the standards' of the field, it can make sense to give them shorter names. Ad-hock labeling data/datatype/datum as d, probably not

s, p, o are very frequently used when you are dealing with RDF triples. Writing d for datatype is an abbreviation (heavily used in owlready2, though).

Reply:
> (...)
> Short line length and use of symbols counts for readability, at least for people with a small tendency of dyslexia. What do you think is most readable
> 
> ```
> E = m * c**2
> ```
> 
> or
> 
> ```
> Energy equals mass times speed of light in vacuum squared
> ```

In this particular example I would prefer the latter, simply due to the inherent semantics provided. That's what's mainly missing when shortening the variables. But a middle-ground would be most preferred for sure.

But your point is taken. Still, care should be taken to write as readable code as possible, by as many developers as possible, within the small sub-set of the overlapping domain knowledge that exists.

Reply:
> > This is a good/interesting point to think about. What is the **scope** of these s, p, o variables? Do they frequently show up in small functions, limited loops, short methods, etc.? If so, and especially if used consistently, and in line with 'the standards' of the field, it can make sense to give them shorter names. Ad-hock labeling data/datatype/datum as d, probably not
> 
> s, p, o are very frequently used when you are dealing with RDF triples. Writing d for datatype is an abbreviation (heavily used in owlready2, though).

Owlready2's Python syntactics should not be repeated outside Owlready2. It is _not_ a model one should try to achieve or immitate.